job-service/src/main/java/com/hpe/caf/services/job/api/filter/JobFilterQuery.java

```diff
@@ -0,0 +1,163 @@
+    private static final DbColumn JOB_PERCENTAGE_COMPLETE;
+
+    private static final DbColumn LABEL_JOB_ID;
+    private static final DbColumn LABEL_LABEL;
+    private static final DbColumn LABEL_PARTITION_ID;
+    private static final DbColumn LABEL_VALUE;
+
+    static {
+        JOB_TABLE = new DbTable(new DbSchema(new DbSpec(), ""), "job", "job");
+        LABEL_TABLE = new DbTable(new DbSchema(new DbSpec(), ""), "label", "label");
+
+        JOB_ID = JOB_TABLE.addColumn("job_id");
+        JOB_NAME = JOB_TABLE.addColumn("name");
+        JOB_PARTITION_ID = JOB_TABLE.addColumn("partition_id");
+        JOB_CREATE_TIME = JOB_TABLE.addColumn("create_date");
+        JOB_LAST_MODIFIED = JOB_TABLE.addColumn("last_update_date");
+        JOB_STATUS = JOB_TABLE.addColumn("job_status");
+        JOB_PERCENTAGE_COMPLETE = JOB_TABLE.addColumn("percentage_complete");
+
+        LABEL_PARTITION_ID = LABEL_TABLE.addColumn("partition_id");
+        LABEL_LABEL = LABEL_TABLE.addColumn("label");
+        LABEL_JOB_ID = LABEL_TABLE.addColumn("job_id");
+        LABEL_VALUE = LABEL_TABLE.addColumn("value");
+
+        COLUMN_MAPPINGS = new ConcurrentHashMap<>();
+        COLUMN_MAPPINGS.put("id", JOB_ID);
+        COLUMN_MAPPINGS.put("name", JOB_NAME);
+        COLUMN_MAPPINGS.put("createTime", JOB_CREATE_TIME);
+        COLUMN_MAPPINGS.put("lastUpdateTime", JOB_LAST_MODIFIED);
+        COLUMN_MAPPINGS.put("status", JOB_STATUS);
+        COLUMN_MAPPINGS.put("percentageComplete", JOB_PERCENTAGE_COMPLETE);
+    }
+
+    public static Condition getQueryStatement(final String key, final ComparisonOperator comparisonOperator, final List<String> args)
+    {
+        if (key.startsWith("labels.")) {
+            final String[] keys = key.split("\\.");
+            final Condition con1 = BinaryCondition.equalTo(JOB_PARTITION_ID, LABEL_PARTITION_ID);
+            final Condition con2 = BinaryCondition.equalTo(JOB_ID, LABEL_JOB_ID);
+            final Condition con3 = convertConditionString(LABEL_LABEL,
+                                                          new ComparisonOperator("=="),
+                                                          Arrays.asList(keys[1]));
+            final Condition con4 = convertConditionString(LABEL_VALUE,
+                                                          comparisonOperator,
+                                                          args);
+            return labelExistsCon(con1, con2, con3, con4);
+
+        }
+        return convertConditionString(COLUMN_MAPPINGS.get(key), comparisonOperator, args);
+    }
+
+    private static Condition convertConditionString(final DbColumn key, final ComparisonOperator comparisonOperator,
+                                                    final List<String> args)
+    {
+
+        final Condition condition;
+        switch (comparisonOperator.getSymbol()) {
+            case "==":
+                condition = BinaryCondition.equalTo(key, args.get(0));

```

Comment:
What about [wildcards](https://github.houston.softwaregrp.net/tlc-apimanagement/rest-standard#wildcards)?  I thought there was something in the FIQL/RSQL docs about asterisks at the beginning or end of the string being treated as wildcards (which is why I wrote that in the REST document).  Wouldn't they need to be converted to a `LIKE` statement in SQL?